pydantic_ai_slim/pydantic_ai/_tool_manager.py

```diff
@@ -121,7 +138,15 @@ async def _call_tool(self, call: ToolCallPart, allow_partial: bool, wrap_validat
             else:
                 args_dict = validator.validate_python(call.args or {}, allow_partial=pyd_allow_partial)
 
-            return await self.toolset.call_tool(name, args_dict, ctx, tool)
+            if usage_limits is not None:
```

Comment:
I think we should also check `count_tool_usage` here, as the "the next tool call would..." error is not correct when the next call is an output tool